kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/Utils.java

```diff
@@ -142,25 +144,36 @@ public static String join(final Object[] array, final char separator) {
   /**
    * Wait until an other thread signals the completion of a task.
    * If an exception is passed, it will be propagated to the caller.
-   * @param queue     The communication channel.
+   * @param future    The communication channel.
    * @param amount    The amount of time to wait.
    * @param timeUnit  The time unit.
    *
    * @return a boolean value indicating resource is ready or not.
    */
-  public static boolean waitUntilReady(BlockingQueue<Object> queue, long amount, TimeUnit timeUnit) {
+  public static boolean waitUntilReady(Future<?> future, long amount, TimeUnit timeUnit) {
```

Comment:
Maybe we should add a note to the Changelog in case someone is using this method

Reply:
I hope they don't either ;) but you never know what users are capable of :man_shrugging: